Clarify update scripts docs on doc vs ctx use (#66357)
elastic/elasticsearch: docs/reference/scripting/fields.asciidoc

`ctx.op`::          The operation that should be applied to the document: `index` or `delete`.
`ctx._index` etc::  Access to <<mapping-fields,document metadata fields>>, some of which may be read-only.

> These scripts do not have access to the `doc` variable and have to use `ctx` to access the documents they operate on.

Reviewer: ```suggestion
Update scripts cannot access the `doc` variable; they must use `ctx` to access the documents they operate on.
```
